build.assets/robotest_run_suite.sh

```diff
@@ -3,17 +3,32 @@ set -eu -o pipefail
 
 readonly UPGRADE_FROM_DIR=${1:-$(pwd)/../upgrade_from}
 
+# decrement_patch returns x.y.(z-1) given valid x.y.z semver.
+decrement_patch() {
```

Comment:
nit: I'm pretty sure you can source this from a separate file to avoid duplication.

Reply:
I won't be returning to this PR, but I will include this guidance in the next edition, as appropriate.